spring-xsuaa/src/test/java/com/sap/cloud/security/xsuaa/token/flows/ClientCredentialsTokenFlowTests.java

```diff
@@ -0,0 +1,237 @@
+		assertThatThrownBy(() -> {
+			new ClientCredentialsTokenFlow(restTemplate, tokenDecoder, null, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageStartingWith("Token");
+
+		assertThatThrownBy(() -> {
+			new ClientCredentialsTokenFlow(restTemplate, tokenDecoder, TestConstants.tokenEndpointUri, null,
+					TestConstants.keySetEndpointUri);
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageStartingWith("Authorize");
+
+		assertThatThrownBy(() -> {
+			new ClientCredentialsTokenFlow(restTemplate, tokenDecoder, TestConstants.tokenEndpointUri,
+					TestConstants.authorizeEndpointUri, null);
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageStartingWith("Key set");
+	}
+
+	@Test
+	public void test_execute_throwsIfMandatoryFieldsNotSet() {
+
+		assertThatThrownBy(() -> {
+			ClientCredentialsTokenFlow tokenFlow = new ClientCredentialsTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(null)
+					.secret(TestConstants.clientSecret)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client ID");
+
+		assertThatThrownBy(() -> {
+			ClientCredentialsTokenFlow tokenFlow = new ClientCredentialsTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.client(TestConstants.clientId)
+					.secret(null)
+					.execute();
+		}).isInstanceOf(IllegalArgumentException.class).hasMessageContaining("client secret");
+
+		assertThatThrownBy(() -> {
+			ClientCredentialsTokenFlow tokenFlow = new ClientCredentialsTokenFlow(restTemplate, tokenDecoder,
+					TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri,
+					TestConstants.keySetEndpointUri);
+			tokenFlow.execute();
+		}).isInstanceOf(TokenFlowException.class).hasMessageContaining("Client credentials flow request is not valid");
+	}
+
+	@Test
+	public void test_execute() throws TokenFlowException {
+
+		HttpEntity<Void> expectedRequest = buildExpectedRequest(clientId, clientSecret);
+
+		URI expectedURI = UriComponentsBuilder.fromUri(TestConstants.tokenEndpointUri)
+				.queryParam("grant_type", "client_credentials").build().toUri();
+
+		RestTemplateMock restTemplateMock = new RestTemplateMock(expectedURI, expectedRequest, Map.class,
+				mockJwt.getTokenValue(), HttpStatus.OK);
+
+		ClientCredentialsTokenFlow tokenFlow = new ClientCredentialsTokenFlow(restTemplateMock, tokenDecoderMock,
+				TestConstants.tokenEndpointUri, TestConstants.authorizeEndpointUri, TestConstants.keySetEndpointUri);
+
+		tokenFlow.client(clientId)
+				.secret(clientSecret)
+				.execute();
+
+		restTemplateMock.validateCallstate();
+		tokenDecoderMock.validateCallstate();
+	}
+
+	@Test
+	public void test_execute_throwsIfHttpStatusUnauthorized() {
```

Comment:
need to fix this test as an unauthorized is represented by a HttpClientErrorException